pkg/controllers/monitoring/monitoring-stack/controller.go

```diff
@@ -143,10 +151,44 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
-			return ctrl.Result{}, err
+			return r.updateConditions(ctx, req, ms)
 		}
 	}
+	return r.updateConditions(ctx, req, ms)
+}
+
+func (r *reconciler) updateConditions(ctx context.Context, req ctrl.Request, ms *stack.MonitoringStack) (ctrl.Result, error) {
+	prom, err := r.mcClient.MonitoringV1().Prometheuses(req.Namespace).Get(ctx, ms.Name, metav1.GetOptions{})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	newStatus := stack.MonitoringStackStatus{}
+	for i := range prom.Status.Conditions {
+		pc := prom.Status.Conditions[i]
+		c := newCondition(pc, ms.Generation)
```

Comment:
nit: Since it's not used anywhere else in the `for` I would do it like this, but I don't have a strong opinion just a suggestion :)
```suggestion
		c := newCondition(prom.Status.Conditions[i], ms.Generation)
```